src/fast-path/__tests__/index.js

```diff
@@ -0,0 +1,569 @@
+      [['//foo/'], '/foo/'],
+      [['//foo', '/'], '/foo/'],
+      [['//foo', '', '/'], '/foo/'],
+      // No UNC path expected (too many leading slashes - questionable)
+      [['///foo/bar'], '/foo/bar'],
+      [['////foo', 'bar'], '/foo/bar'],
+      [['\\\\\\/foo/bar'], '/foo/bar'],
+      // Drive-relative vs drive-absolute paths. This merely describes the
+      // status quo, rather than being obviously right
+      [['C:'], 'C:.'],
+      [['c:'], 'c:.'],
+      [['c:.'], 'c:.'],
+      [['c:', ''], 'c:.'],
+      [['', 'c:'], 'c:.'],
+      [['c:.', '/'], 'c:./'],
+      [['c:.', 'file'], 'c:file'],
+      [['c:', '/'], 'c:/'],
+      [['c:', 'file'], 'c:/file'],
+    ]);
+
+    it('should join the paths correctly (posix)', () => {
+      const fp = require('../index');
+
+      posixJoinTests.forEach((test) => {
+        const actual = fp.join.apply(fp, test[0]);
+        const expected = test[1];
+
+        expect(actual).toEqual(expected, [
+          `fp.join(${test[0].map(JSON.stringify).join(',')})`,
+          `expect=${JSON.stringify(expected)}`,
+          `actual=${JSON.stringify(actual)}`,
+        ].join('\n'));
+      });
+    });
+
+    it('should join the paths correctly (win32)', () => {
+      process.platform = 'win32';
+      const fp = require('../index');
+
+      win32JoinTests.forEach((test) => {
+        const actual = fp.join.apply(fp, test[0]);
+        const expected = test[1].replace(/\//g, '\\');
+
+        expect(actual).toEqual(expected, [
+          `fp.join(${test[0].map(JSON.stringify).join(',')})`,
+          `expect=${JSON.stringify(expected)}`,
+          `actual=${JSON.stringify(actual)}`,
+        ].join('\n'));
+      });
+    });
+
+    it('should throw for invalid input', () => {
+      const fp = require('../index');
+      invalidInputTests.forEach((test) => {
+        expect(() => fp.join(test)).toThrow();
+      });
+    });
+  });
+
+  describe('normalize', () => {
+    it('should return a valid path (posix)', () => {
+      const fp = require('../index');
+
+      expect(fp.normalize('./fixtures///b/../b/c.js')).toEqual('fixtures/b/c.js');
+      expect(fp.normalize('/foo/../../../bar')).toEqual('/bar');
+      expect(fp.normalize('a//b//../b')).toEqual('a/b');
+      expect(fp.normalize('a//b//./c')).toEqual('a/b/c');
+      expect(fp.normalize('a//b//.')).toEqual('a/b');
+    });
+
+    it('should return a valid path (win32)', () => {
+      process.platform = 'win32';
+      const fp = require('../index');
+
+      expect(fp.normalize('./fixtures///b/../b/c.js')).toEqual('fixtures\\b\\c.js');
+      expect(fp.normalize('/foo/../../../bar')).toEqual('\\bar');
+      expect(fp.normalize('a//b//../b')).toEqual('a\\b');
+      expect(fp.normalize('a//b//./c')).toEqual('a\\b\\c');
+      expect(fp.normalize('a//b//.')).toEqual('a\\b');
+      expect(fp.normalize('//server/share/dir/file.ext')).toEqual('\\\\server\\share\\dir\\file.ext');
+    });
+  });
+
+  describe('resolve', () => {
```

Comment:
awesome work on making the tests work both ways.

Here are a few more suggestions:
* you can actually do `const fp = require('../');` on top of the file. We use the inline-requires transform to replace every occurrence of `fp` with `require('../')`. Since we reset the module registry, this works quite well for individual calls to `it`. As long as the first use of `fp` is after accessing `platform`, it should work fine :)
* There is quite a few copy-pasta in this test file now, here is what I would recommend:

```js
const win32Tests = […];
const posixResolveTests = [];
const runResolveTests = (platform, tests) => {
  it('works with the tests', () => {
    process.platform = platform;
    tests.forEach(test => {
       const actual = fp.resolve.apply(fp, test[0]);
       const expected = test[1];
 
       expect(actual).toEqual(expected, [
         `fp.resolve(${test[0].map(JSON.stringify).join(',')})`,
         `expect=${JSON.stringify(expected)}`,
         `actual=${JSON.stringify(actual)}`,
       ].join('\n'));
     });
  });
}
```